webview/src/shared/hooks/useVsCodeMessaging.ts

```diff
@@ -10,13 +10,13 @@ const signalInitialized = () =>
   sendMessage({ type: MessageFromWebviewType.INITIALIZED })
 
 export function useVsCodeMessaging<T extends WebviewData>(
-  handler: (event: { data: MessageToWebview<T> }) => void
+  handler?: (event: { data: MessageToWebview<T> }) => void
 ) {
   useEffect(() => {
     signalInitialized()
   }, [])
   useEffect(() => {
-    window.addEventListener('message', handler)
-    return () => window.removeEventListener('message', handler)
+    handler && window.addEventListener('message', handler)
+    return () => handler && window.removeEventListener('message', handler)
```

Comment:
[Q] Any objections to this? I don't need to send this webview any data right now, it is either on or off. This will change if/when we expand the webview to connect to other products.